### ISSUE TITLE
Inquiry regarding num_train_epochs=70 for GPT2-XL

### ISSUE THREAD

**Issue by junle-chen** [LUMIA-Group/MemoryDecoder#6]
Hi,

I noticed that num_train_epochs is set to 70 in scripts/train_memdec.sh. This seems quite high for fine-tuning a model as large as GPT2-XL (1.5B).

Could you please clarify the training time: Roughly how long did the full training take in your experiments?

Looking forward to your reply.

**Comment by mxmcao** [comment]
Hi, thank you for your interest.

The long epoch training is only used for gpt-2 small size(124M) MemDec in our paper. As for 0.5B size MemDec, we train for roughly 15 epochs, which is comparable to 1-epoch training on a 7B model as stated in our paper.

During our preliminary research, we find that the distribution alignment task is hard specifically for smaller models. Loss continues to drop even after 100 epochs without overfitting, while other parametric methods quickly overfit.
Larger models learn quicker and you can notice considerable improvements after a few epochs.

I hope the explanation above answers your question.

**Comment by mxmcao** [comment]
FYI: We use gpt2-xl to build knn datastore but the MemDec itself is trained from gpt2-small
